package/spec/frontend/media_player/hooks_spec.js

```diff
@@ -64,27 +62,31 @@ describe('hooks', function() {
         expect(player.intendingToPause()).toBe(false);
       });
 
-      it('calls original play method when promise returned by before is resolved', function() {
+      it('calls original play method when promise returned by before is resolved', async () => {
         var player = fakePlayer();
-        var deferred = new jQuery.Deferred();
+
+        var promiseResolve;
+        var promise = new Promise(function(resolve, reject) {
+          promiseResolve = resolve;
+        });
+
         hooks(player, {
           before: function() {
-            return deferred.promise();
+            return promise;
           }
         });
 
         player.play();
-        deferred.resolve();
+        await promiseResolve();
 
         expect(player.originalPlay).toHaveBeenCalled();
       });
 
       it('does not call original play method until promise returned by before is resolved', function() {
         var player = fakePlayer();
-        var deferred = new jQuery.Deferred();
         hooks(player, {
           before: function() {
-            return deferred.promise();
+            return Promise.resolve();
           }
         });
 
```

Comment:
This test now looks a bit misleading. Before the `before` promise stayed pending. Now `originalPlay` _would_ be called, it is just not called synchronously, which makes the test pass,

We could consider rejecting the promise instead and expecting that play also rejects and orginal play is not called.